### PR TITLE
Remove unneeded framework files

### PR DESCRIPTION
**Pull request by schlessera**
As the tests package require several WP-CLI commands, and therefore also directly requires the WP-CLI framework, there's no point in having a (currently unsynced) copy of the three framework files `utils.php`, `Process.php` and `ProcessRun.php` included with the tests package.

This PR removes theses files, and makes the bootstrapping code more robust to find the version of the files that comes with the framework.

A side benefit is that there's only 1 single location now for these files, and all packages will always be up-to-date without additional PRs.

Depends on #53

Related #44

### CODE REVIEW

**Comment by schlessera** [comment]
This is technically a breaking change, as the files that were initially included are now gone.

However, I doubt that any third-party package directly relied on them being in that specific location. As long as they are loaded and usable, everything should still work as expected.

**Comment by schlessera** [comment]
Additional note: the PHP Unit tests were already using the `utils.php` from the framework dependency, instead of the one that was included with this package. This means that up until now, the helper functions used in PHP Unit were not the same as the ones used in the Behat tests. This could have caused (or actually did) hard to diagnose bugs.

File: features/bootstrap/FeatureContext.php [hunk 1]
```diff
@@ -6,39 +6,90 @@
...
+		dirname( dirname( __DIR__ ) ) . '/vendor',
+		// wp-cli/wp-cli-tests is a dependency.
+		dirname( dirname( dirname( __DIR__ ) ) ),
+	];
+
+	$vendor_folder = '';
+	foreach ( $vendor_locations as $location ) {
+		if ( is_dir( $location )
+		     && is_readable( $location )
+		     && is_file( "{$location}/autoload.php" ) ) {
```
Comment (jrfnl) [inline features/bootstrap/FeatureContext.php]: These two lines have space based precision alignment, please adjust this to ordinary tab indents.
```suggestion
			&& is_file( "{$location}/autoload.php" ) ) {
```

File: features/bootstrap/FeatureContext.php [hunk 2]
```diff
@@ -6,39 +6,90 @@
...
+		$project_folder,
+		// wp-cli/wp-cli is a dependency.
+		"{$vendor_folder}/wp-cli/wp-cli",
+	];
+
+	$framework_folder = '';
+	foreach ( $framework_locations as $location ) {
+		if ( is_dir( $location )
+		     && is_readable( $location )
+		     && is_file( "{$location}/php/utils.php" ) ) {
```
Comment (jrfnl) [inline features/bootstrap/FeatureContext.php]: These two lines have space based precision alignment, please adjust this to ordinary tab indents.

File: features/bootstrap/FeatureContext.php [hunk 3]
```diff
@@ -6,39 +6,90 @@
...
+
+	$contents = "require:\n";
+	foreach ( $composer->autoload->files as $file ) {
+		$contents .= "  - {$project_folder}/{$file}\n";
 	}
+
+	$temp_folder = sys_get_temp_dir() . '/wp-cli-package-test';
+	if ( ! is_dir( $temp_folder )
+	     && ! mkdir( $temp_folder )
+	     && ! is_dir( $temp_folder ) ) {
```
Comment (jrfnl) [inline features/bootstrap/FeatureContext.php]: These two lines have space based precision alignment, please adjust this to ordinary tab indents.

File: features/bootstrap/FeatureContext.php (hunk 1 above)
Comment (jrfnl) [reply features/bootstrap/FeatureContext.php]: Also for readability/ to distinguish more easily between the conditions and the code within the control structure, you may want to consider writing this like so:
```php
		if ( is_dir( $location )
			&& is_readable( $location )
			&& is_file( "{$location}/autoload.php" )
		) {
			$vendor_folder = $location;
			break;
		}
```

Comment (schlessera) [reply features/bootstrap/FeatureContext.php]: They are not supposed to be indented. All three conditions are at the same indentation level, they are equals.

Comment (jrfnl) [reply features/bootstrap/FeatureContext.php]: > They are not supposed to be indented. All three conditions are at the same indentation level, 

Except they aren't, the second and third condition are indented by an additional five spaces (which should be one tab).

> they are equals.

That I get, but is not the point.

Comment (schlessera) [reply features/bootstrap/FeatureContext.php]: the five spaces are alignment, like when you align assignments, only here we align boolean comparison operators. That's a common practice.

Comment (schlessera) [reply features/bootstrap/FeatureContext.php]: Like here: https://github.com/php-fig-rectified/fig-rectified-standards/blob/master/PSR-2-R-coding-style-guide-additions.md#multi-line-declarationconditionconcatenation

Comment (schlessera) [reply features/bootstrap/FeatureContext.php]: > Except they aren't, the second and third condition are indented by an additional five spaces (which should be one tab).

Spaces are not indentation. All three conditions are at the very same indentation level, with two of them having additional alignment (but not additional indentation). Changing the spaces into a tab turns them from alignment to indentation, so we 1 at level 3 and 2 at level 4.

Comment (schlessera) [reply features/bootstrap/FeatureContext.php]: Ok, how about this:
```php
		if (
			is_dir( $location )
			&& is_readable( $location )
			&& is_file( "{$location}/autoload.php" )
		) {
			$vendor_folder = $location;
			break;
		}
```
That keeps them at the same indentation, but avoids the spaces.

Comment (schlessera) [reply features/bootstrap/FeatureContext.php]: Hmm, I thought to remember something with regards to this discussion...
https://github.com/squizlabs/PHP_CodeSniffer/pull/1736

Comment (jrfnl) [reply features/bootstrap/FeatureContext.php]: I hear you, but the WP Coding Standards does not allow for alignment at the start of a line, only for indentation.

> Rule of thumb: Tabs should be used at the beginning of the line for indentation, while spaces can be used mid-line for alignment.

Ref: https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#indentation

> Ok, how about this:

That works :+1: (and as a side-bonus complies with PSR-12).

**Pull request merged**